src/main/java/org/radarbase/appserver/service/questionnaire/protocol/GithubProtocolFetcherStrategy.java

```diff
@@ -259,17 +266,11 @@ private Map<String, URI> getProtocolDirectories() throws IOException {
     }
 
     private Protocol getProtocolFromUrl(URI uri) throws IOException, InterruptedException {
-        HttpResponse response = client.send(getRequest(uri), HttpResponse.BodyHandlers.ofString());
-        if (isSuccessfulResponse(response)) {
-            final ObjectMapper mapper =
-                    objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-            GithubContent content = mapper.readValue(uri.toURL(), GithubContent.class);
-            return mapper.readValue(content.getContent(), Protocol.class);
-        } else {
-            log.error("Error getting Protocol from URL {} : {}", uri.toString(), response);
-            throw new ResponseStatusException(
-                    HttpStatus.valueOf(response.statusCode()), "Protocol could not be retrieved");
-        }
+        String contentString = this.githubClient.getGithubContent(uri.toString());
+        final ObjectMapper mapper =
+                objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
Would this not change the behaviour of the injected mapper (and could have unintended consequence in other places that it is used)?

I would create a new ObjectMapper at the top of the class as copy of this one like

```
private final ObjectMapper localMapper;
```

Then init it in the constructor - 
```
this.localMapper = new ObjectMapper(objectMapper);
this.localMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```